src/WireMock.Net/ResponseBuilders/Response.cs

```diff
@@ -285,7 +305,10 @@ public IResponseBuilder WithProxy(string proxyUrl, string clientX509Certificate2
 
             if (ProxyUrl != null)
             {
-                return await HttpClientHelper.SendAsync(requestMessage, ProxyUrl, X509Certificate2Filename);
+                var requestUri = new Uri(requestMessage.Url);
```

Comment:
Why is this needed?

Why not just add the new parameter (X509Certificate2Password) to the method?